precompiles/distribution/tx_test.go

```diff
@@ -354,6 +354,19 @@ func (s *PrecompileTestSuite) TestClaimRewards() {
 			true,
 			"invalid type for maxRetrieve: expected uint32",
 		},
+		{
+			"fail - too many retrieved results",
+			func() []interface{} {
+				return []interface{}{
+					s.address,
+					uint32(32_000_000),
+				}
+			},
+			func([]byte) {},
+			200000,
+			true,
+			"maxRetrieve parameter exceeds the number of validator",
+		},
 		{
 			"success - withdraw from all validators - 2",
```

Comment:
Nit: add a test case when the maxRetrieve is > than delegator validators but is a valid number.